recipes/sol2/3.x.x/conanfile.py

```diff
@@ -0,0 +1,69 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class Sol2Conan(ConanFile):
+    name = "sol2"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ThePhD/sol2"
+    description = "C++17 Lua bindings"
+    topics = ("conan", "lua", "c++", "bindings")
+    settings = "os", "compiler"
+    license = "MIT"
+    requires = ["lua/5.3.5"]    
+    default_options = {"lua:compile_as_cpp": True}
```

Comment:
Why is this required?
Imho, it's unexpected behavior if some default_option is changed by a dependent package.

People can always set `lua:compile_as_cpp=False`, so this (failure?) mode is still possible.

Reply:
It should compile as C by default. Lua as C++ will probably trip up most people. Unless...

Does the lua/5.3.5 compile as C++, or C, by default? It matters for interop.

Reply:
Does my comment at the define in `package_info` help things?

Reply:
since sol is a C++ library, I thought that is the logical choice. 
lua/5.3.5 has an option if it is C++ or not, and this can be set in a profile
and it can be queried, and then the sol build could adjust
and its not hard to add, but I needed a starting point to open the discussion so here it is.
and to me the C++ default seamed OK, especially since I have a concrete use case for this :-)